### PR TITLE
Added python bindings for rgbd module

### PR DESCRIPTION
**Pull request by Piggelinus**
### This pullrequest changes

Exposes RGBD Odometry classes and functions to Python

### CODE REVIEW

**Comment by berak** [comment]
hmm, did you check, if this is producing working python bindings locally on your box ?
(i'd think, it would need `static Ptr<MyClass>::create()` functions here, not public class constructors)

the [whitespace issue](http://pullrequest.opencv.org/buildbot/builders/precommit-contrib_docs/builds/10771/steps/whitespace%20opencv_contrib/logs/stdio) should be easy to fix.

**Comment by Piggelinus** [comment]
Yeah, I've got RgbdOdometry working on correctly on the TUM freiburg1_xyz dataset in Python 2.7 on Ubuntu 16.04.

Do you mean adding a static Ptr<MyClass>::create() method to all classes instead of wrapping the constructors directly? That does indeed seem like the way it's done in other modules.

**Comment by berak** [comment]
> working on correctly on the TUM

oh, apologies for the noise, then, misread it.

**Comment by Piggelinus** [comment]
Should I make the changes I mentioned?

File: modules/rgbd/include/opencv2/rgbd.hpp [hunk 1]
```diff
@@ -56,13 +56,13 @@ namespace rgbd
...
    * @param depth the depth to check for validity
    */
-  CV_EXPORTS
+  CV_EXPORTS_W
   inline bool
   isValidDepth(const float & depth)
   {
     return !cvIsNaN(depth);
   }
-  CV_EXPORTS
```
Comment (sovrasov) [inline modules/rgbd/include/opencv2/rgbd.hpp]: Do we really need `isValidDepth*` functions to be exported? I think `nan` checks can be done in Python.

File: modules/rgbd/include/opencv2/rgbd.hpp [hunk 2]
```diff
@@ -81,7 +81,7 @@ namespace rgbd
     return (depth != std::numeric_limits<unsigned short int>::min())
         && (depth != std::numeric_limits<unsigned short int>::max());
   }
-  CV_EXPORTS
```
Comment (sovrasov) [inline modules/rgbd/include/opencv2/rgbd.hpp]: `isValidDepth3` is a strange name, it's better to use something like `isValidDepthInt` to make code readable.

File: modules/rgbd/include/opencv2/rgbd.hpp [hunk 3]
```diff
@@ -238,44 +243,46 @@ namespace rgbd
...
     ~DepthCleaner();
 
+    CV_WRAP static Ptr<DepthCleaner> create(int depth, int window_size = 5, int method = DepthCleaner::DEPTH_CLEANER_NIL);
+
     /** Given a set of 3d points in a depth image, compute the normals at each point.
      * @param points a rows x cols x 3 matrix of CV_32F/CV64F or a rows x cols x 1 CV_U16S
      * @param depth a rows x cols matrix of the cleaned up depth
      */
-    void
+    CV_PROP void
```
Comment (sovrasov) [inline modules/rgbd/include/opencv2/rgbd.hpp]: As far as I know `CV_PROP` allows to wrap simple public data members like
`CV_PROP int a;`
does this operator really works from python?

Comment (Piggelinus) [inline modules/rgbd/include/opencv2/rgbd.hpp]: No, I see what you mean. I couldn't get it to work with CV_WRAP either so I'll remove the bindings for operator().

**Comment by sovrasov** [comment]
:+1:

**Pull request merged**